Formula/m/malcontent.rb

```diff
@@ -0,0 +1,22 @@
+class Malcontent < Formula
+  desc "Supply Chain Attack Detection, via context differential analysis and YARA"
+  homepage "https://github.com/chainguard-dev/malcontent"
+  url "https://github.com/chainguard-dev/malcontent/archive/refs/tags/v1.6.0.tar.gz"
+  sha256 "a8538888caee8d16fce62c398fc8eaa1af9aca225139afe9c8247c84202403db"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+  depends_on "yara"
+
+  on_linux do
+    depends_on "pkgconf" => :build
+  end
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w", output: bin/"mal"), "./cmd/mal"
+  end
+
+  test do
+    assert_match "malcontent version", shell_output("#{bin}/mal --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate tenew-formula-rapidyamlath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@chenrui333 Thanks - I considered what might work here, and since this tool is mainly to discover and audit malicious code behaviors, I didn't think it would be a good idea to create that for testing purposes.

Would a test that emits "nothing found" be more valid for the test here?

Reply:
Thanks - I think I have a good-enough test now, let me know what you think.